### DISCUSSION TITLE
How to update the netbox device data from external data by using python

### DISCUSSION THREAD

**Discussion by selvigp** [netbox-community/netbox#15400] [Help Wanted!]
I have an external file, needs to validate those data with netbox data. If there is any mismatch, want to update that data into netbox_data. I have tried to use `setattr(device, field, value)` , but somehow its not updating instead it returns None

**Comment by candlerb** [comment]
Without seeing your code, I can't speculate as to what you've done wrong. It's not even clear if you're using the Django ORM (e.g. in a custom script) or the REST API.

For the Django ORM there are working example scripts and reports you can copy in the https://github.com/netbox-community/customizations repo, and you may want to look at the [Django ORM documentation](https://docs.djangoproject.com/en/4.2/topics/db/queries/#saving-changes-to-objects). In short, the general pattern is:

```
obj = ... retrieve object from database ...
obj.snapshot()   # for netbox changelog
... update the obj properties ...
obj.full_clean() # don't omit this, otherwise no model validation is done
obj.save()
```
